chat-sdk-ui/src/main/java/co/chatsdk/ui/ImageListAdapter.java

```diff
@@ -0,0 +1,107 @@
+package co.chatsdk.ui;
+
+import android.net.Uri;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+
+public class ImageListAdapter extends RecyclerView.Adapter<ImageListAdapter.MyViewHolder>  {
+
+    public ArrayList<Uri> uriArrayList = new ArrayList<>();
+    private WeakReference<SendMultipleImagesAtOnceActivity> theMainActivity;
+
+    public ImageListAdapter (SendMultipleImagesAtOnceActivity activity) {
+        this.theMainActivity = new WeakReference<SendMultipleImagesAtOnceActivity>(activity);
+    }
+
+    public class MyViewHolder extends RecyclerView.ViewHolder {
+
+        private WeakReference<ImageListAdapter> adapter;
+        public ImageView imageView;
+        public Uri thisImageUri;
+        public int thisImagePosition;
+
+        public MyViewHolder(ImageListAdapter la, ImageView v) {
+            super(v);
+            this.adapter = new WeakReference<ImageListAdapter>(la);
+            imageView = v;
+
+            imageView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    setMainImageDisplay(thisImageUri);
+                    setMainImageUri(thisImageUri);
+                }
+            });
+
+            imageView.setOnLongClickListener(new View.OnLongClickListener() {
+                @Override
+                public boolean onLongClick(View v) {
+                    Uri toBeDeletedUri = thisImageUri;
+                    removeImageFromAdapter(thisImageUri);
+                    notifyDataSetChanged();
+                    if (getMainImageUri() == toBeDeletedUri) {
+                        if (uriArrayList.size() >= 1 && uriArrayList.size() == thisImagePosition) {
```

Comment:
1. Add a comment to explain this line. 
2. `.size() > 0` instead of `.size() >= 1`
3. This will always return false `uriArrayList.size() == thisImagePosition`. Can you explain what it's supposed to do?

Reply:
Also, you can't access `uriArrayList` here. Call `adapter.get().uriArrayList` and if you want to know the position of an item use `uriArrayList.indexOf(uri)`

Reply:
That second point is a problem, it has to do with the issue at the top that I left unresolved.